test/Microsoft.AspNetCore.Routing.Tests/Matchers/MatcherConformanceTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    public abstract class MatcherConformanceTest
```

Comment:
The 'matcher' contract is a little more rich than the routing contracts of yore, so it seems good to express these tests in terms of 'matcher' rather than anything else.